### PR TITLE
Improve based on perlcritic feedback

### PR DESCRIPTION
**Pull request by mtompset**
- no tabs
- no post-fix ifs
- no funky variables ($0 -> $PROGRAM_NAME via English)
- no die (use Carp qw(croak);)
- extracted the !$quiet logic to a function
- moved functions to the end
- removed unnecessary bracketing on built-in functions
- added xlsm to various regexp's

### CODE REVIEW

**Comment by devon-sil** [comment]
@dalenewby Can we get this reviewed?

File: youtrack-backup.pl [hunk 1]
```diff
@@ -94,12 +89,12 @@ ()
...
-printf "%s: Backing up YouTrack database\n", tstamp() if (! $quiet);
+output_if_not_quiet("Backing up YouTrack database\n");
 $result = `$curl_cmd | jq '.backupStatus.backupError'`;
 
-chomp($result);
-if ($result ne "null") {
-	die "Database backup failed: $result\n";
+chomp $result;
+if ( $result ne 'null' ) {
+    croak "Database backup failed: $result\n";
```
Comment (jason-jackson) [inline youtrack-backup.pl:97]: I think this one should still be `die`. See https://www.perlmonks.org/?node_id=685452

> You use die when the error is something you or your code didn't do right. You use croak when it's something your caller isn't doing right.

File: youtrack-backup.pl [hunk 2]
```diff
@@ -159,10 +152,10 @@ ()
...
 $curl_cmd   = "curl $accept_header $auth_header $progress_header $curl_query";
 
-printf "%s: Downloading file $download_path\n", tstamp() if (! $quiet);
-$result = system($curl_cmd);
-if ($result != 0) {
-	die "Download failed: $result\n";
+output_if_not_quiet("Downloading file $download_path\n");
+$result = system $curl_cmd;
+if ( $result != 0 ) {
+    croak "Download failed: $result\n";
```
Comment (jason-jackson) [inline youtrack-backup.pl:158]: I think this should remain `die` too (see above)

File: youtrack-backup.pl [hunk 3]
```diff
@@ -178,18 +171,32 @@ ()
...
-$result     = system($cmd);
-if ($result != 0) {
-	`cat $tmpfile`;
-	unlink($download_path);
-	die "Upload failed: $result\n";
+$result     = system $cmd;
+if ( $result != 0 ) {
+    `cat $tmpfile`;
+    unlink $download_path;
+    croak "Upload failed: $result\n";
```
Comment (jason-jackson) [inline youtrack-backup.pl:181]: I think this should remain `die` too (see above)

File: youtrack-backup.pl [hunk 4]
```diff
@@ -178,18 +171,32 @@ ()
...
+sub tstamp() {
+    return strftime '%H:%M:%S', gmtime;
+}
+
+sub output_if_not_quiet {
+    my ($output) = @_;
+
+    if ( !$quiet ) {
+        $time_stamped_output = sprintf "%s: %s", tstamp(), $output;
+        print $time_stamped_output;
```
Comment (jason-jackson) [inline youtrack-backup.pl:199]: It seems odd to me to sprintf something to only print it the next line.

```suggestion
        printf "%s: %s", tstamp(), $output;
```

**Review by jason-jackson — COMMENTED** [review]
My perl knowledge is a bit rusty, haven't used it in over 10 years, but had a couple comments.

**Comment by mtompset** [comment]
No point. I give up. croak() doesn't trigger perlcritic errors like die() does. Also, stacktrace is nice.

**Pull request closed without merging**